frontend-opal/src/components/PortfolioContainer/AllocationInfo/AllocationInfoContainer.js

```diff
@@ -0,0 +1,62 @@
+import { connect } from 'react-redux';
+import palette from 'google-palette';
+import getTotalInvestment from 'components/common/getTotalInvestment';
+import presenter from './AllocationInfoPresenter';
+
+const mapStateToProps = ({ portfolio: { instruments } }) => {
+  let sectors = {};
+  let totalInvestment = '';
+  if (instruments) {
```

Comment:
@WenyuD these two ifs is actually one.
and they wouldn't be needed at all, if you don't have this weird state:
```javascript
  state: {
    watchlists: ...,
    portfolio: {
      instruments: [...]
    }
  }
```
you can just have:
```javascript
  state: {
    watchlists: ...,
    portfolio: [...]
  }
```